.github/workflows/build.yml

```diff
@@ -0,0 +1,80 @@
+name: Build Binaries
+
+on:
+  workflow_call:
+    inputs:
+      ref:
+        type: string
+      addon-name:
+        type: string
+        default: hex_map
+    outputs:
+      artifact-name:
+        value: ${{ inputs.addon-name }}
+
+env:
+  # Use UTF-8 on Linux.
+  LANG: en_US.UTF-8
+  LC_ALL: en_US.UTF-8
+
+jobs:
+  build:
+    name: Build GDExtension
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          [
+            { platform: linux, arch: x86_64, os: ubuntu-22.04 },
+            { platform: windows, arch: x86_64, os: windows-latest },
+            { platform: windows, arch: x86_32, os: windows-latest },
+            { platform: macos, arch: universal, os: macos-latest },
+            { platform: android, arch: arm64, os: ubuntu-22.04 },
+            { platform: android, arch: arm32, os: ubuntu-22.04 },
+            { platform: android, arch: x86_64, os: ubuntu-22.04 },
+            { platform: android, arch: x86_32, os: ubuntu-22.04 },
+            { platform: ios, arch: arm64, os: macos-latest },
+            { platform: web, arch: wasm32, os: ubuntu-22.04 }
+          ]
+        target-type: [editor, template_debug, template_release]
+        # disabling double precision builds for now; doubles the CI time for
+        # unlikely case.
+        float-precision: [single] # , double]
+
+    runs-on: ${{ matrix.target.os }}
+    env:
+      BUILD_VERSION: ${{ matrix.target.platform }}_${{ matrix.target.arch }}_${{ matrix.float-precision }}_${{ matrix.target-type }}
+      SCONS_CACHE: ${{ github.workspace }}/.scons-cache/
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref }}
+          submodules: true
+
+      - name: Setup godot-cpp build environment
+        uses: godotengine/godot-cpp/.github/actions/setup-godot-cpp@godot-4.4-stable
+        with:
+          platform: ${{ matrix.target.platform }}
+          em-version: 3.1.62
+
+      - name: Restore Scons cache
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.SCONS_CACHE }}
+          key: scons_cache-${{ env.BUILD_VERSION }}_${{ hashFiles('.gitmodules') }}
+
+      - name: Build GDExtension
+        shell: sh
+        run: |
+          scons \
+            verbose=1 \
+            target=${{ matrix.target-type }} \
+            platform=${{ matrix.target.platform }} \
+            arch=${{ matrix.target.arch }} \
+            precision=${{ matrix.float-precision }}
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ inputs.addon-name }}-${{ env.BUILD_VERSION }}_${{ hashFiles('.gitmodules') }}
+          path: ${{ github.workspace }}/demo/addons/${{ inputs.addon-name }}/lib
```

Comment:
You removed the merge build artifacts step here.  Please put it back in.  It makes it easier to pull the built PR for testing without doing a release.  This isn't a release archive, it's an archive of all the built DLLs.

```yml
  # Merges all the build artifacts together into a single godot-hex-map artifact.
  # If you comment out this step, all the builds will be uploaded individually.
  merge:
    name: merge build artifacts
    needs: build
    runs-on: ubuntu-22.04
    steps:
      - name: Merge Artifacts
        uses: actions/upload-artifact/merge@v4
        with:
          name: godot-hex-map
          pattern: godot-hex-map-*
          delete-merged: true
```

Note: with the windows cleanup step, you shouldn't need the `merge-multiple: true` anymore.

Reply:
The artifacts were being merged at point of download instead (in both release and CI). Not sure it makes much difference, and to your previous comment - would make it possible to download only the artifacts under test. Happy to change at your preference though.